File: java/learning-spring-02/src/main/java/com/example/learningspring/beans/game/items/MarioGame.java
package com.example.learningspring.beans.game.items;

import org.springframework.context.annotation.Primary;
import org.springframework.stereotype.Component;

@Component
@Primary
public class MarioGame implements GameConsole {
    public void up() {
        System.out.println("up");
    }

    public void down() {
        System.out.println("Sit down");
    }

    public void left() {
        System.out.println("Go back");
    }

    public void right() {
        System.out.println("Shoot a bullet");
    }
}

File: kotlin/learning-spring/src/main/kotlin/com/examplekotlin/learningspring/coupled/loosely/game/MarioGame.kt
package com.examplekotlin.learningspring.coupled.loosely.game

class MarioGame : GameConsole {
    override fun up() {
        println("up")
    }

    override fun down() {
        println("Sit down")
    }

    override fun left() {
        println("Go back")
    }

    override fun right() {
        println("Shoot a bullet")
    }
}